queries/dune_v2/period_block_interval.sql

```diff
@@ -1,6 +1,16 @@
 -- V3: https://dune.com/queries/1541504
-select
-    min("number") as start_block,
-    max("number") as end_block
-from ethereum.blocks
-where date_trunc('minute', time) between cast('{{StartTime}}' as timestamp) and cast('{{EndTime}}' as timestamp)
+with block_range_start as (
+    select
+        max("number") as start_block
+    from ethereum.blocks
+    where time <= cast('{{StartTime}}' as timestamp)
```

Comment:
I would expect the minimal change would be

`date_trunc('minute', time) >= cast('{{StartTime}}' as timestamp) and date_trunc('minute', time) <= cast('{{EndTime}}' as timestamp)`

I am not at all comfortable with all these casts, timestamps etc so don't get me wrong, I am not making a case against what you did, just observing.

Reply:
With the current code there can be overlap between block ranges from adjacent weeks. In the worst case this can lead to double counting of auctions (if they happen at exactly the right time).

![image](https://github.com/cowprotocol/solver-rewards/assets/5250107/ff819606-4799-4c36-8c50-bf1cf4040148)

An example of an overlap can be seen here:
- week ending on 2.1. has an `end_block` of `18916006`: https://dune.com/queries/3330918?StartTime_d83555=2023-12-26+00%3A00%3A00&EndTime_d83555=2024-01-02+00%3A00%3A00
- week starting on 2.1. has a `start block` of `18916002`: https://dune.com/queries/3330918?StartTime_d83555=2024-01-02+00%3A00%3A00&EndTime_d83555=2024-01-05+00%3A00%3A00

Note that the [query](https://dune.com/queries/1541504) actually called by the code does not truncate. It was probably missed to update it. Without truncation there is a different problem with inconsistent behavior between block times being exactly aligned with the dates or being slightly off.

The updated code in this PR should always select the same block as end in week n and start in week n+1.